api/management/commands/triggers_to_db.py

```diff
@@ -73,5 +73,54 @@ def handle(self, *args, **options):
 AND a.user_id                 = b.user_id
 AND coalesce(a.event_id  , 0) = coalesce(b.event_id   , 0);
 COMMIT;
+
+
+CREATE OR REPLACE FUNCTION log_api_appeal_changes()
+RETURNS TRIGGER AS $$
+DECLARE datNow date = now();
+BEGIN
+
+   -- close current period
+   UPDATE api_appealhistory SET valid_to = datNow where aid = OLD.aid; 
+   
+   -- insert a new period
+   INSERT INTO api_appealhistory(appeal_id,valid_from, valid_to, num_beneficiaries, amount_requested, amount_funded, aid, start_date, end_date,atype, country_id, region_id)
+   VALUES (NEW.id,datNow,'2200-01-01',NEW.num_beneficiaries,NEW.amount_requested,NEW.amount_funded,NEW.aid, NEW.start_date, NEW.end_date,NEW.atype, NEW.country_id, NEW.region_id);
+
+   RETURN NEW;
+END;
+$$ language 'plpgsql';
+
+CREATE OR REPLACE FUNCTION log_api_appeal_insterts()
+RETURNS TRIGGER AS $$
+DECLARE datNow date = now();
+BEGIN
+
+   -- insert a new period
+   INSERT INTO api_appealhistory(appeal_id,valid_from, valid_to, num_beneficiaries, amount_requested, amount_funded, aid, start_date, end_date,atype, country_id, region_id)
+   VALUES (NEW.id,datNow,'2200-01-01',NEW.num_beneficiaries,NEW.amount_requested,NEW.amount_funded,NEW.aid, NEW.start_date, NEW.end_date,NEW.atype, NEW.country_id, NEW.region_id);
+
+   RETURN NEW;
+END;
+$$ language 'plpgsql';
+
+
+
+DROP TRIGGER IF EXISTS tu_api_appeal_01 on api_appeal;
+CREATE TRIGGER tu_api_appeal_01 BEFORE UPDATE
+ON api_appeal FOR EACH ROW
+-- we update real_data_update time only if important figures has been changed:
+    WHEN ((OLD.num_beneficiaries IS DISTINCT FROM NEW.num_beneficiaries)
+       OR (OLD.amount_requested  IS DISTINCT FROM NEW.amount_requested)
+       OR (OLD.amount_funded     IS DISTINCT FROM NEW.amount_funded))
+    EXECUTE PROCEDURE log_api_appeal_changes();
+
+
+DROP TRIGGER IF EXISTS ti_api_appeal_01 on api_appeal;
+CREATE TRIGGER ti_api_appeal_01 BEFORE INSERT
+ON api_appeal FOR EACH ROW
+    EXECUTE PROCEDURE log_api_appeal_insterts();
+
```

Comment:
Let's use the Django signal for this. (in `api/receivers.py`)

Here is a sample.
```python3
@receiver(post_save, sender=Appeal)
def add_update_appeal_history(sender, instance, created, **kwargs):
    fields_watched = [
        field.name for field in AppealHistory._meta.get_fields()
        if field.name not in ['id', 'appeal', 'valid_from', 'valid_to']
    ]
    now = timezone.now()
    latest_attributes = {
        field: getattr(instance, field)
        for field in fields_watched
    }.update(dict(
        appeal=instance,
        valid_from=now,
        # valid_to=  # Let's leave this to null and when we filter in the views.py lets coalesce this with now date
    ))

    if created:  # Appeal Insert
        AppealHistory.objects.create(**latest_attributes)
    else:  # Appeal Update
        appeal_history = AppealHistory.objects.filter().order_by('id').last()
        if appeal_history and not any([getattr(instance, field) != getattr() for field in fields_watched]):
            # Watched fields are not changed
            return
        appeal_history.valid_to = now
        appeal_history.save(update_fields='valid_to')
        # Create a new appeal history if changed
         AppealHistory.objects.create(**latest_attributes)
```